recipes/openvino/all/conandata.yml

```diff
@@ -0,0 +1,34 @@
+sources:
+  "2023.1.0":
+    "openvino":
+      url: "https://github.com/openvinotoolkit/openvino/archive/47b736f63edda256d66e2bbb572f42a9d6549f6e.tar.gz"
+      sha256: "8a1dd5b35fe4b60440e97ee61e12e01aa4dd78b5be30550aa08fcd96c416e1d7"
+    "arm_compute":
+      url: "https://github.com/ARM-software/ComputeLibrary/archive/refs/tags/v23.02.1.tar.gz"
+      sha256: "c3a443e26539f866969242e690cf0651ef629149741ee18732f954c734da6763"
+    "onednn_cpu":
+      url: "https://github.com/openvinotoolkit/oneDNN/archive/ae825539bd850d1ad5c83d4bb0d56c65d46d5842.tar.gz"
+      sha256: "1204df17785c8603f9dfa1f4f91e91e5ffd4391bf7680d2b256de2513490ebee"
+    "mlas":
+      url: "https://github.com/openvinotoolkit/mlas/archive/c7c8a631315000f17c650af34431009d2f22129c.tar.gz"
+      sha256: "7b790dfeef8e1dd612f920c85186c52ad3a3e2245e2a2afd6cc91ce4b1dc64a9"
+    "onednn_gpu":
```

Comment:
I could be wrong (maintainer please chime in here!), but my understanding is that dependencies can't be vendored as source code like this for CCI (assuming I understand your intent here correctly). My understanding is that these need to be modelled as requirements, so each would need its own recipe.

Additionally, I'm not sure what the current state of oneDNN is - it looks like it might be open source now? At one point it was closed source or its license was prohibitive, and so it got a special mention in https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#how-to-package-libraries-that-depend-on-proprietary-closed-source-libraries

If it is still closed source or proprietary, I don't think it would be packageable on CCI.

Reply:
> My understanding is that these need to be modelled as requirements, so each would need its own recipe.

I agree about oneDNN for GPU and ARM Computer Library, because they are official repositories. But there are not such recipes in Conan for now and it's a simplest way to enable OpenVINO recipe.

In future, once me or someone else will add such recipes, we can re-use it here. But I suppose it's not a blocker to merge current recipe. 

> Additionally, I'm not sure what the current state of oneDNN is - it looks like it might be open source now? At one point it was closed source or its license was prohibitive

It's Apache 2.0, just check "about" section of the repo https://github.com/oneapi-src/oneDNN

Some others dependencies (like MLAS) are pure OpenVINO internal code, which is just located in a submodule for OpenVINO repo for better modularity. But we don't have a goal to distribute it as a separate product / component.

Reply:
> In future, once me or someone else will add such recipes, we can re-use it here. But I suppose it's not a blocker to merge current recipe.

It might be a blocker - I've had PRs blocked on this in the past. A maintainer will have to make a decision on that.

> It's Apache 2.0, just check "about" section of the repo https://github.com/oneapi-src/oneDNN

Yes, I think what's confused me in the past has been that for some of the one* libraries the interfaces were open source but the implementations weren't. I think this applied to oneMKL but it wasn't clear to me how this applied to other libraries in the family.

> Some others dependencies (like MLAS) are pure OpenVINO internal code, which is just located in a submodule for OpenVINO repo for better modularity. But we don't have a goal to distribute it as a separate product / component.

That's good context to have, thanks for clarifying. Does the MLAS dependency you've got share any commonality with https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/core/mlas?

[This statement](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#handling-internal-dependencies) indicates that it might need a package to avoid ODR violations if symbols haven't been deconflicted with upstream. There's no package in CCI for mlas at the moment so I'm not sure how much of a risk this poses, but there is one for onnxruntime, so maybe there's a chance for conflicts?

Reply:
> so maybe there's a chance for conflicts?

Yes, but OpenVINO and ONNX Runtime are both for the same tasks, so customers usually have to select which inference framework to use and go to production only with one of them. So, on practice such conflict is unlikely to happen.

Reply:
I've slotted this topic for our Monday/Tuesday meeting, thanks for your patience :)

Reply:
Ok, note, that we are OK to improve the recipe over the time. 
Our vision is to split the work into parts. We want to add OpenVINO - let's add OpenVINO first, its dependencies later, not in opposite way :)
Currently, those dependencies are supposed to be submodules. So, for 2023.1 it's the only way. But once ARM Compute Library / oneDNN recipe is out, we can add such type of source into our cmake infra and consume packages from Conan cache as well.

Reply:
I am perfectly fine with the maintainers doing a release of ``openvino`` based on 3 different sources repositories, with their own commits, etc. That is not vendoring of dependencies, they are releasing their ``openvino`` the way they intend to release it and be used by consumers of it. The fact that there are 3 different repos is an internal implementation detail, and it is not fair or necessary to force them to do separate package releases for every source repo. If they chose to do it in the future, that is good, but shouldn't be mandated from Conan side. 

OneDNN coming from the same org, is a bit in the limit of the vendoring definition, but coming from the same org, I am also fine with initially allowing it, and postponing a bit the extraction as independent package, being this the recommended path.

A different story is the ``arm_compute``  package, that looks like a vendoring of a third party. Even if the result is fully embedded and isolated, at the very least it could be causing duplication of efforts for someone else that might want to use ``arm_compute``. I'd recommend to extract this one in its own package, as we have been doing in the past, unless there are some big challenges or blockers to make ``arm_compute`` its own package, if you need help or can't do it now, we might be able to try to help and contribute it.

(Note: If we have had cases in the past where authors and library maintainers were not allowed to vendor their own organization sources and forced to release independent packages, we should revisit that, learned why it happened, and try to improve and clarify for the future. There are many things in ConanCenter that are not immutable truths, things evolve with the experience, knowledge and changes in the community and ecosystem)

Reply:
> I'd recommend to extract this one in its own package

I agree with such position, but am I right that it's not a blocker to merge? 
It can be done later as improvement. 

(BTW, I have already made 19 PRs to CCI which got merged, so I want to finally merge our recipe and continue improvements after merge)